modules/sfp_badpackets.py

```diff
@@ -0,0 +1,246 @@
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+        
+        if self.errorState:
+            return None
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        # Always check if the API key is set and complain if it isn't, then set
+        # self.errorState to avoid this being a continual complaint during the scan.
+        if self.opts['api_key'] == "":
+            self.sf.error("You enabled sfp_badpackets but did not set an API key!", False)
+            self.errorState = True
+            return None
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+        else:
+            self.results[eventData] = True
+
+        if eventName == 'NETBLOCK_OWNER':
+            if not self.opts['netblocklookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxnetblock']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxnetblock']))
+                    return None
+        
+        if eventName == 'NETBLOCK_MEMBER':
+            if not self.opts['subnetlookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxsubnet']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxsubnet']))
+                    return None
+                    
+        qrylist = list()
+        if eventName.startswith("NETBLOCK_"):
+            for ipaddr in IPNetwork(eventData):
+                qrylist.append(str(ipaddr))
+                self.results[str(ipaddr)] = True
+        else:
+            # If user has enabled affiliate checking
+            if eventName == "AFFILIATE_IPADDR" and not self.opts['checkaffiliates']:
+                return None
+            qrylist.append(eventData)
+        
+        for addr in qrylist:
+
+            nextPageHasData = True    
+            if self.checkForStop():
+                return None
+            
+            currentOffset = 0
+            while nextPageHasData:
+                data = self.queryIPAddress(addr, currentOffset)
+
+                if data is None:
+                    nextPageHasData = False
+                    break
+
+                count = data.get('count')
+                if count is None or int(count) == 0:
+                    nextPageHasData = False
+                    break
+
+                # Data is returned about the IP Address
+                evt = SpiderFootEvent("IP_ADDRESS", addr, self.__name__, event)
```

Comment:
You should not generate this event if the incoming event was `IP_ADDRESS` - it should only be generated when the incoming event was a `NETBLOCK_*`.